scipy/spatial/meson.build

```diff
@@ -0,0 +1,59 @@
+qhull_src = ['qhull_src/src/poly_r.c',
+'qhull_src/src/stat_r.c',
+'qhull_src/src/merge_r.c',
+'qhull_src/src/userprintf_r.c',
+'qhull_src/src/userprintf_rbox_r.c',
+'qhull_src/src/usermem_r.c',
+'qhull_src/src/geom_r.c',
+'qhull_src/src/libqhull_r.c',
+'qhull_src/src/user_r.c',
+'qhull_src/src/random_r.c',
+'qhull_src/src/geom2_r.c',
+'qhull_src/src/mem_r.c',
+'qhull_src/src/global_r.c',
+'qhull_src/src/poly2_r.c',
+'qhull_src/src/io_r.c',
+'qhull_src/src/rboxlib_r.c',
+'qhull_src/src/qset_r.c']
+
+# TODO: Add c_args: numpy_nodepr_api, once NumPy API usage is corrected.
+qhull = py3.extension_module('qhull',
+  ['qhull.pyx', 'qhull_misc.h', 'qhull_misc.c'] + qhull_src,
+  include_directories: [incdir_numpy, '../_lib', '../_build_utils/src', 'qhull_src/src'],
+  dependencies: [py3_dep],
+  install : true,
+  subdir : 'scipy/spatial')
+
+ckdtree_src = ['ckdtree/src/query.cxx',
+  'ckdtree/src/build.cxx',
+  'ckdtree/src/query_pairs.cxx',
+  'ckdtree/src/count_neighbors.cxx',
+  'ckdtree/src/query_ball_point.cxx',
+  'ckdtree/src/query_ball_tree.cxx',
+  'ckdtree/src/sparse_distances.cxx']
+
+ckdtree_cpp = custom_target('ckdtree_cpp',
+  output : 'ckdtree.cpp',
+  input : 'ckdtree.pyx',
+  command : [cython, '--cplus', '-3', '--fast-fail', '@INPUT@', '-o', '@OUTPUT@']
+)
+
+# TODO: Add cpp_args: numpy_nodepr_api, once NumPy API usage is corrected.
```

Comment:
I've started using this line instead of a comment, keeping the TODO but silencing the warnings in the meantime:
```python
cpp_args: '-Wno-cpp',  # TODO: replace with numpy_nodepr_api
```

For `c_args` I haven't checked the equivalent flag to add, but I guess there is one.

Reply:
I found, `-w` which disables all the compiler warnings. I am not sure if it is what we want.

Reply:
That seems unhealthy:) What I'm seeing is:
```
[20/33] Compiling C object scipy/spatial/_distance_wrap.cpython-39-x86_64-linux-gnu.so.p/src_distance_wrap.c.o
In file included from /home/rgommers/anaconda3/envs/scipy-meson/lib/python3.9/site-packages/numpy/core/include/numpy/ndarraytypes.h:1944,
                 from /home/rgommers/anaconda3/envs/scipy-meson/lib/python3.9/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                 from /home/rgommers/anaconda3/envs/scipy-meson/lib/python3.9/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from ../scipy/spatial/src/distance_wrap.c:49:
/home/rgommers/anaconda3/envs/scipy-meson/lib/python3.9/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
   17 | #warning "Using deprecated NumPy API, disable it with " \
      |  ^~~~~~~
```

so it looks like `c_args: '-Wno-cpp` works for C code too.